gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/GenerateMetricSchemaTask.java

```diff
@@ -45,8 +58,14 @@ public final void generate() {
         JavaGenerator.generate(JavaGeneratorArgs.builder()
                 .inputs(getSource().getFiles().stream().map(File::toPath).collect(ImmutableSet.toImmutableSet()))
                 .output(output.toPath())
+                .libraryName(Optional.ofNullable(libraryName.getOrNull()))
                 // TODO(forozco): probably want something better
                 .defaultPackageName(getProject().getGroup().toString())
                 .build());
     }
+
+    private String defaultLibraryName() {
+        String rootProjectName = getProject().getRootProject().getName();
+        return rootProjectName.replaceAll("-root$", "");
```

Comment:
I really think its lame to rely on a convention on a seemingly unrelated project to determine the name. Could we require users to specify the field if they wanted to opt in?

Reply:
When we discussed in the standup, it sounded like we wanted to design this so that library authors who are _not_ us would be likely to get it right. I actually want to keep this functionality as an 'on by default' thing otherwise I don't think any metric-schema users apart from us will flip this switch.

Interestingly the main libraries we care about don't actually have the `-root` suffix. I could just delete it and say if you don't like the default you're free to override?
- c-j-r https://github.com/palantir/conjure-java-runtime/blob/develop/settings.gradle#L1
- dialogue https://github.com/palantir/dialogue/blob/develop/settings.gradle#L1
- atlasdb https://github.com/palantir/atlasdb/blob/develop/settings.gradle#L1
- tritium https://github.com/palantir/tritium/blob/develop/settings.gradle#L1

Reply:
> On the other hand it would be odd if two components from the same repository had different library names, right?

It might be odd, but it could also make sense in especially large repos that produce multiple libraries/distributions.

> I actually want to keep this functionality as an 'on by default' thing otherwise I don't think any metric-schema users apart from us will flip this switch.

Agreed that it makes sense for libraries to have this on by default. I disagree that it makes sense for _all_ users of metric-schema to produce a `libraryName`, `libraryVersion` tag. For services, its at best a bit confusing and at worst we're adding unnecessary strain on the logging infrastructure by producing duplicate information.

Reply:
I just prefer these things to be a bit more explicit and I'm willing trade off the cost/risk of asking libraries to opt in vs. producing unnecessary information for services since libraries are quite uncommon and in general we don't encourage/allow just anyone to write them

Reply:
@ferozco I'm not convinced by those reasons. I hold the on-by-default thing as _really_ valuable, and I don't see how any this will cause any confusion if the tag gets added a bit superfluously to servers. We've added tags to stuff in the past without telling people, I don't see how this is different. Similarly, I don't think adding a few more bytes to each log line is gonna flatten the signals infra but if it does then I'd be ok with flipping it round.